ARExperiment/ARExperiment/RecognizeObjectsViewController.swift

```diff
@@ -100,68 +109,150 @@ class RecognizeObjectsViewController: UIViewController, ARSCNViewDelegate {
+            showStartButtonIfError()
+            print("we have no model")
             return
         }
+        node.position = SCNVector3(pointTranslation.x, pointTranslation.y, pointTranslation.z)
+        sceneView.scene.rootNode.addChildNode(node)
+    }
 
+    private func nodeForTextAndSize() -> SCNNode? {
         guard let model = arViewModel.createSceneNodeForAsset(nodeName,
                                                               fileName: fileName,
                                                               assetExtension: fileExtension) else {
-            print("we have no model")
-            return
+                                                                showStartButtonIfError()
+                                                                print("we have no model")
+                                                                return nil
         }
+        let parentNode = SCNNode()
+        let text = SCNText(string: objectName, extrusionDepth: 0.5)
+        text.styleFirstMaterial(with: UIColor.blue)
+        let textNode = SCNNode(geometry: text)
+        textNode.scale = SCNVector3(textNode.scale.x * 0.01 ,
+                                    textNode.scale.y * 0.01 ,
+                                    textNode.scale.z * 0.01)
+        let modelWidth = model.boundingBox.max.x - model.boundingBox.min.x
+        let modelHeight = model.boundingBox.max.y - model.boundingBox.min.y
+        let widthScale = Float(boundingBoxSize.width) / modelWidth
+        let heightScale = Float(boundingBoxSize.height) / modelHeight
+        model.scale = SCNVector3(model.scale.x * widthScale,
+                                 model.scale.y * heightScale,
+                                 model.scale.z)
+        parentNode.addChildNode(model)
+        parentNode.addChildNode(textNode)
+        return parentNode
+    }
+}
 
-        compoundingBox.frame = scaledRect
-        predictionLabel.text = "\(labels[prediction.classIndex])"
-        compoundingBox.isHidden = false
-
-        let scaledPoint = CGPoint(x: scaledRect.origin.x, y: scaledRect.origin.y)
-        if let hitPoint = arViewModel.hitResult(at: scaledPoint, in: sceneView, withType: [.existingPlaneUsingExtent, .estimatedHorizontalPlane]) {
-            let pointTranslation = hitPoint.worldTransform.translation
-            model.position = SCNVector3(pointTranslation.x, pointTranslation.y, pointTranslation.z)
-            sceneView.scene.rootNode.addChildNode(model)
 
+extension RecognizeObjectsViewController: ARSCNViewDelegate {
+    func renderer(_ renderer: SCNSceneRenderer, nodeFor anchor: ARAnchor) -> SCNNode? {
+        guard !anchor.isKind(of: ARPlaneAnchor.self) && usingAnchors else {
+            return nil
         }
+        guard let model = nodeForTextAndSize() else {
+            showStartButtonIfError()
+            print("we have no model")
+            return nil
+        }
+        return model
```

Comment:
this file is getting very long - consider splitting stuff out, if you can (e.g. node construction)

Reply:
Yes, it has been also reduced in this PR -> https://github.com/novoda/ios-demos/pull/36 but will take a second look too here see what i can move